pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -176,31 +181,37 @@ class Win32DeploymentReader {
     let locked: RecursivePartial<settings.Settings> = {};
 
     this.errors = [];
-    for (const keyName of ['HKLM', 'HKCU'] as const) {
-      this.keyName = keyName;
-      const key = nativeReg[keyName];
-      const registryKey = nativeReg.openKey(key, this.registryPathProfile.join('\\'), nativeReg.Access.READ);
-
-      if (!registryKey) {
-        continue;
-      }
-      const defaultsKey = nativeReg.openKey(registryKey, DEFAULTS_HIVE_NAME, nativeReg.Access.READ);
-      const lockedKey = nativeReg.openKey(registryKey, LOCKED_HIVE_NAME, nativeReg.Access.READ);
-
-      try {
-        defaults = defaultsKey ? this.readRegistryUsingSchema(settings.defaultSettings, defaultsKey, [DEFAULTS_HIVE_NAME]) : {};
-        locked = lockedKey ? this.readRegistryUsingSchema(settings.defaultSettings, lockedKey, [LOCKED_HIVE_NAME]) : {};
-      } catch (err) {
-        console.error('Error reading deployment profile: ', err);
-      } finally {
-        nativeReg.closeKey(registryKey);
-        nativeReg.closeKey(defaultsKey);
-        nativeReg.closeKey(lockedKey);
-      }
-      // If we found something in the HKLM Defaults or Locked registry hive, don't look at the user's
-      // Alternatively, if the keys work, we could break, even if both hives are empty.
-      if (Object.keys(defaults).length || Object.keys(locked).length) {
-        break;
+    // eslint-disable-next-line no-labels
+    readProfileFromRegistry: {
+      for (this.registryPathProfileIndex = 0; this.registryPathProfileIndex < this.registryPathProfile.length; this.registryPathProfileIndex++ ) {
+        for (const keyName of ['HKLM', 'HKCU'] as const) {
+          this.keyName = keyName;
+          const key = nativeReg[keyName];
+          const registryKey = nativeReg.openKey(key, this.registryPathProfile[this.registryPathProfileIndex].join('\\'), nativeReg.Access.READ);
```

Comment:
With the change above this simply becomes

```suggestion
          const registryKey = nativeReg.openKey(key, this.registryPath.join('\\'), nativeReg.Access.READ);
```

Reply:
changed.